lib/modules/manager/pipenv/extract.ts

```diff
@@ -39,17 +35,19 @@ function extractFromSection(
       let currentValue: string | undefined;
       let nestedVersion = false;
       let skipReason: SkipReason | undefined;
-      if (requirements.git) {
-        skipReason = 'git-dependency';
-      } else if (requirements.file) {
-        skipReason = 'file-dependency';
-      } else if (requirements.path) {
-        skipReason = 'local-dependency';
-      } else if (requirements.version) {
-        currentValue = requirements.version;
-        nestedVersion = true;
-      } else if (is.object(requirements)) {
-        skipReason = 'unspecified-version';
+      if (is.object(requirements)) {
```

Comment:
why this new check? didn't you already validate in isRequirement?

Reply:
`requirements` can be either a string or an object from the PipFile, so I updated the type definition to go along with the method that validated it's type. TypeScript complains if one tries to use a potential-string as an object, hence putting all the object-assuming code inside an `is.object()` check.